src/display/graphics.cpp

```diff
@@ -1625,7 +1625,19 @@ double Graphics::getScale() const {
 
 void Graphics::setScale(double factor) {
     p->threadData->rqWindowAdjust.wait();
-    factor = clamp(factor, 0.5, 4.0);
+    
+    int displayIndex = SDL_GetWindowDisplayIndex(shState->sdlWindow());
+    SDL_Rect displayRect;
+    SDL_GetDisplayUsableBounds(displayIndex, &displayRect);
+    
+    int top, bottom, left, right;
+    SDL_GetWindowBordersSize(shState->sdlWindow(), &top, &bottom, &left, &right);
+    
+    double maxWidth = displayRect.w;
+    double maxHeight = displayRect.h - top;
```

Comment:
Why are you only using `top` rather than all four outputs of `SDL_GetWindowBordersSize`?